Dockerfile

```diff
@@ -57,6 +57,10 @@ ADD scripts/controlflow.properties $GEOSERVER_DATA_DIR
 ADD scripts/sqljdbc4-4.0.jar $CATALINA_HOME/webapps/geoserver/WEB-INF/lib/
 
 RUN /scripts/setup.sh \
+    && groupadd -r geoserver && useradd --no-log-init -r -g geoserver geoserver \
+    && chown --verbose --recursive geoserver:geoserver /opt/geoserver \
+    && chown --verbose --recursive geoserver:geoserver /opt/footprints_dir \
+    && chown --verbose --recursive geoserver:geoserver /usr/local/tomcat \
```

Comment:
@EHJ-52n What is the difference between adding a user with a UID like
```
#change Container from Root User to geoserver
RUN groupadd -r geoserver -g 10001 && \
    useradd -M -u 10000 -g geoserver geoserver
RUN chown -R geoserver:geoserver /usr/local/tomcat && \
    chmod -R 511 $CATALINA_HOME/webapps/geoserver/WEB-INF/lib/
```

Reply:
In terms of docker, each `RUN`, `ADD`, `USER`, `EXPOSE`,... command provides an additional layer which I would like to avoid when creating docker images.

In terms of user ids, what is the reason to use such high ids? Do you want to prevent using already existing ids of the docker host system?

In terms of chmod, why is mode `511` required for `geoserver/WEB-INF/lib/`?

Reply:
I really do not know much about running containers as none root user so that's why I was just asking. I read here https://medium.com/@mccode/understanding-how-uid-and-gid-work-in-docker-containers-c37a01d01cf
https://medium.com/better-programming/running-a-container-with-a-non-root-user-e35830d1f42a